### PR TITLE
Refresh the sign-up and auth reset flow UIs

### PR DESCRIPTION
**Pull request by bl-nero**
This PR implements layout changes and splits the WebAuthn credential creation in 2 steps, as required by the [UI design](https://www.figma.com/file/v6GunK50D2VC7w7I2FBDNf/Management?node-id=3707%3A36633&mode=dev).

There will be a follow-up PR with changes to the account recovery UI.

Demo: https://zoom.us/clips/share/_yAHIlAXn3qtdY66KWx9v3QTgPQcuoe_xWLbXHzCS1rbCu-Q9eAy8ZtqFcUAKZkJmanMDaQlkqOBDvxQNb_sKyi6.y-CDdEDBEP-osiXG

Changelog: Changed UI for the sign-up and authentication reset flows.

### CODE REVIEW

**Comment by bl-nero** [comment]
Tested following cases:

- Setting up / resetting an account
	- Passwordless default (OSS)
	- Passwordless default, picked password (OSS)
	- Username/password + MFA default (OSS, WebAuthn)
	- Username/password + MFA default (OSS, OTP)
	- Username/password + MFA default, picked passwordless (OSS)
	- Username/password + MFA default (Enterprise)
	- MFA optional, no MFA
	- MFA off
- Adding a WebAuthn MFA method through Account Settings (doesn't work because of a separate issue, see https://github.com/gravitational/teleport/issues/40528)

File: web/packages/teleport/src/Welcome/NewCredentials/NewMfaDevice.tsx [hunk 1]
```diff
@@ -144,94 +144,95 @@ export function NewMfaDevice(props: NewMfaDeviceProps) {
...
                 />
-                <Text
-                  fontSize={1}
-                  textAlign="center"
-                  mt={2}
-                  color="text.slightlyMuted"
-                >
-                  Scan the QR Code with any authenticator app and enter the
-                  generated code. We recommend{' '}
-                  <Link href="https://authy.com/download/" target="_blank">
```
Comment (zmb3) [inline web/packages/teleport/src/Welcome/NewCredentials/NewMfaDevice.tsx:171]: Strange, I could swear you already removed this recently. 🤷

**Review by zmb3 — APPROVED** [review]
Looks great, thanks for the detailed test cases too!

**Review by avatus — APPROVED** [review]

File: web/packages/teleport/src/Welcome/NewCredentials/NewMfaDevice.tsx (hunk 1 above)
Comment (bl-nero) [reply web/packages/teleport/src/Welcome/NewCredentials/NewMfaDevice.tsx:171]: I did, on a different component :)

**Pull request merged**